Create a new line with zero-width space when pressing enter after a list

simbas opened the pull request:
issue 19611: http://support.web-education.net/issues/19611

Proposed solution:
When I press enter in the last item of a list, if the item is empty the editor override the default browser behavior and adds a `<div>&#8203;</div>` after the list.

simbas committed:
fix(editor): #19611 create a new line with zero-width space after list

- override the default browser behaviour when pressing enter on a last
and empty list item
- remove magic numbers in onPressEnter file
- add unit tests for onPressEnter function

simbas committed:
fix(editor): #19611 fix a compatibility issue with ie11

fixed ie11 issues:
- document.createTreeWalker needs all arguments (no optional argument)
- parentElement is returning undefined (probably when the parent is not
a element, so now we loop over parentNode to find the closest
parentElement)

rdjedjig reviewed (approved):
Je comprends bien le correctif. En revanche Faut soit commenter la méthode onPressEnter, soit la découper. Sans tester de manière avancée, j'ai du mal à bien me figurer tout les cas de traitement

simbas committed:
test(editor): test pressing enter in empty editor

simbas committed:
docs(editor)

simbas committed:
test(editor): system test about double enter problem with empty editor

simbas committed:
fix(editor): issue 19611, selection fixes on delete and empty editor

- move onPressDelete to a separated file
- remove a selection change when pressing delete in an unempty line
- adds a selection change when pressing enter in an empty editor

simbas committed:
refactor(editor): little refactor + fix typo in pressDelete tests
